Enhancement: Added Sliding Toggle Animation to Theme Switcher

codevory opened the pull request:
### **Enhancement: Added Sliding Toggle Animation to Theme Switcher**

Fixes #123 

This PR improves the existing theme switcher by adding a smooth sliding toggle animation, making the interaction more intuitive and visually engaging.

While working on this feature, I noticed that a basic theme switcher had already been implemented. Instead of duplicating functionality, I focused on enhancing the user experience by introducing a sliding effect that better communicates the theme change (light ↔ dark).

As this is my first open-source contribution, it took me around 5–6 hours to understand the codebase, align with the existing implementation, and integrate the animation cleanly.

I hope this improvement adds value to the UI/UX.
Feedback, suggestions, or improvements are very welcome , I’d love to learn and iterate further!


<img width="661" height="917" alt="Screenshot 2025-12-29 144344" src="https://github.com/user-attachments/assets/185dfa26-d7fa-4a7c-98b8-604d2b2ddf6a" />
`
<img width="672" height="917" alt="Screenshot 2025-12-29 144400" src="https://github.com/user-attachments/assets/4a877726-c816-47ef-bcf3-407c0d79028d" />
`


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added a new GlassSwitch toggle component for accessible, stylized switch interactions.

* **Refactor**
  * ThemeSwitch now uses GlassSwitch, adopting switch semantics (role="switch", aria-label/aria-checked) and thumb-based icon rendering.

* **Documentation**
  * New GlassSwitch docs and updated ThemeSwitch docs with usage, accessibility, examples, and tests.

* **Tests**
  * Added GlassSwitch tests and updated ThemeSwitch tests to reflect switch semantics.

* **Chores**
  * Added prop-types and adjusted dev/test dependencies.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Ryan-Millard reviewed (changes requested):
Hi @codevory. Thank you for your contribution - the community appreciates it!🦔

This is very nice, but it appears to have a few bugs and doesn't fully align with #123. I'll explain why below:

### Bugs & Inconsistencies
The `ThemeSwitch` component appears to display both `lucide-react` icons regardless of the current theme and uses the incorrect backgrounds between the two themes:

<img width="120" height="67" alt="Image" src="https://github.com/user-attachments/assets/937334e3-2375-4d19-a7fe-c43f9012090b" />

<img width="111" height="66" alt="Image" src="https://github.com/user-attachments/assets/4f6a830e-1d31-4980-8cf9-e036ecff09e4" />

Additionally (as you can see), the icons don't fit in the switch properly. The fix for that will be discussed in the next section.👇

### Alignment with #123 
#123 specifically asks for a **reusable `GlassSwitch`** - this is because it will allow the switch to be used anywhere inside the application, thereby decoupling it from specific implementations (such as with the `ThemeSwitch`) and reducing code repetition.

The bugs the current implementation currently faces could be fixed more easily through a modular `GlassSwitch` component since individual tests can be written for it and bugs can be addressed more easily since it will be a simpler component.

## Summary of Requested Changes
1. Please create a new file, `GlassCard.jsx`, inside `src/components` and use the global `glass` class, which can be found inside [`src/global-styles/components.css`](https://github.com/Ryan-Millard/Img2Num/blob/main/src/global-styles/components.css).
    - It would be a good idea to have a look at the existing glass components (such as [`GlassCard`](https://github.com/Ryan-Millard/Img2Num/blob/main/src/components/GlassCard.jsx)) to understand how to implement this `GlassSwitch`.
    - Using the global styles will prevent CSS conflicts and ensure the component fits in well with the rest of the UI. This is the fix for the background problem.
2. If you are comfortable with it and have created the `GlassSwitch`, please write some:
    - Tests for the new `GlassSwitch` component (in a file next to `GlassSwitch.jsx`, called `GlassSwitch.test.jsx`) to ensure it functions properly (we use Vitest).
    - Documentation in the `docs/docs` folder (it uses Docusaurus's MDX, which is a blend between Markdown and React).

## Ask Questions
I understand that you're new to open-source, so it would make sense if you got stuck or were confused by something. I'm here to help if you need me and you can also ask @coderabbit for help (it's an LLM that is familiar with this repository) - you've probably already seen the reviews it leaves on pull requests.

Be careful with @coderabbit, though - it is AI, meaning that it can be, and is often, wrong, so use your own judgement to make sure it isn't recommending something over-the-top or completely wrong.

There is also a [Q&A discussion](https://github.com/Ryan-Millard/Img2Num/discussions/24) if you'd like help from the community.

## Additional Resources
### General
I'm sure you've already been through the [introduction page](https://ryan-millard.github.io/Img2Num/info/docs/category/-introduction/) that helps with setting up your local environment. After that, you should make sure to read through [Img2Num's guidelines ](https://ryan-millard.github.io/Img2Num/info/docs/category/guidelines/). 

### Documentation
We have [documentation on how to write documentation in Docusaurus](https://ryan-millard.github.io/Img2Num/info/docs/category/%EF%B8%8F-writing-documentation/), but that could be a bit too minimal depending on your circumstances. In that case, the [official Docusaurus documentation](https://docusaurus.io/docs) is the best place to go if you need help with writing docs.

### Tests
[Vitest's documentation](https://vitest.dev/guide/) is the best place to go for this since Img2Num does not have a guide on how to write tests yet. @coderabbit can also help you with writing tests if you need.







I'm sorry for writing so much - I hope it wasn't a problem. Thanks again for the contribution and please feel free to ask for help when you need it. Have a good day!🦔🦔

codevory commented:
Hi @Ryan-Millard 👋

I’ve made the requested changes and would appreciate a review when you have time.
What I’ve done
- Created a reusable GlassSwitch component that is decoupled from theme logic.
- Added a sliding animation to the switch itself.
- Replaced the previous theme switch button usage with GlassSwitch.
- Used the existing theme hook to control theme state (no theme logic embedded inside GlassSwitch).
- Added tests for GlassSwitch (Vitest) to cover accessibility and interaction.
<hr>

**What I did not change**

- The original ThemeSwitch files remain unaltered.
- No breaking changes to existing theme logic or hooks.

The GlassSwitch is now fully reusable and can be used for other toggles (e.g. mobile menu, feature flags) without modification.
Please let me know if this aligns with the intent of #123, or if you’d like anything adjusted .
Thanks again for the guidance and review! 🙏

Ryan-Millard reviewed (dismissed):
This is exactly what we needed. Thank you!

I left individual comments on some of the files in this PR, so please will you address them as well as the below general ones:

- Please add some documentation in the `docs/` folder
- Fix the formatting inconsistencies in your code (like the `GlassSwitch` props and the CSS file).

Ryan-Millard reviewed (changes requested):
I'm sorry for accidentally approving the PR in the previous review - that was a mistake.

Please see the previous review I left for the changes I'd like you to implement.

Thank you again for the great work!

codevory commented:
Thanks for the summary that matches my intent perfectly.

The goal was to keep GlassSwitch purely presentational and reusable, while preserving ThemeSwitch as the owner of theme-related logic per SRP and issue #123.

Here are screenshots:
<img width="134" height="90" alt="image" src="https://github.com/user-attachments/assets/5627280e-6dfe-467f-9a9b-05845e3ce76f" />
<img width="132" height="91" alt="image" src="https://github.com/user-attachments/assets/ef4535b1-d3d4-47f3-bc7c-3533ba464ff7" />


Happy to make any further adjustments if needed.

Ryan-Millard commented on src/components/GlassSwitch.jsx:
This is great! The propTypes are perfect and everything works well. I'd just like one more thing to be done:

<img width="531" height="153" alt="Image" src="https://github.com/user-attachments/assets/17c717d7-7016-496b-a44f-94166db654e5" />

In the `GlassSwitch` above, I didn't pass `thumbContent` and it didn't display a fallback. Please will you change it to use a default thumb that is the round circle this component had when I last reviewed it - it was very nice. It may not even need to change dynamically, I recommend seeing what it looks like with the `thumbContent` inside of it (it could still look good).

```jsx
<GlassSwitch isOn={true} onChange={() => {}} ariaLabel="hfd" />
```

Ryan-Millard commented on src/components/ThemeSwitch.jsx:
Please add the comments back to this file. They are important (especially the JSDoc comments).

Ryan-Millard reviewed (commented):
This is great. Thank you! It's almost ready to be merged.

Please address the individual comments I left on the files as well as:
- Write documentation for the `GlassSwitch`'s tests
- Update the [documentation for `ThemeSwitch`](http://localhost:3000/Img2Num/info/docs/reference/react/components/ThemeSwitch/) 
- Fix the test failures on `ThemeSwitch` (since `ThemeSwitch` has pretty much completely changed) as well as [its documentation](http://localhost:3000/Img2Num/info/docs/reference/react/components/ThemeSwitch/tests/)
    ```bash
    [14:46] ~/projects/Img2Num $ ./img2num npm run test
    
    > imgtonum@0.0.0 test
    > vitest run
    
    
     RUN  v4.0.16 /usr/src/app
    
     ❯ src/components/ThemeSwitch.test.jsx (7 tests | 7 failed) 3349ms
         × renders a button with the correct aria-label when theme is light 1833ms
         × shows Moon icon for light theme and applies icon class 31ms
         × shows Sun icon for dark theme and applies icon class 24ms
         × calls toggleTheme when clicked 579ms
         × is keyboard accessible (focusable) 380ms
         × works with unexpected/falsy theme values (defaults to Moon icon for non-"dark") 19ms
         × uses the toggleTheme coming from the hook 465ms
    stderr | src/components/GlassSwitch.test.jsx > GlassSwitch > is keyboard accessible
    An update to q inside a test was not wrapped in act(...).
    
    When testing, code that causes React state updates should be wrapped into act(...):
    
    act(() => {
      /* fire events that update state */
    });
    /* assert on the output */
    
    This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
    An update to ForwardRef inside a test was not wrapped in act(...).
    
    When testing, code that causes React state updates should be wrapped into act(...):
    
    act(() => {
      /* fire events that update state */
    });
    /* assert on the output */
    
    This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
    
     ✓ src/components/GlassSwitch.test.jsx (5 tests) 5304ms
         ✓ renders a switch button  1469ms
         ✓ calls onChange when clicked  2289ms
         ✓ is keyboard accessible  1155ms
     ✓ src/components/Tooltip.test.jsx (5 tests) 8002ms
         ✓ does not show tooltip content by default  1398ms
         ✓ shows tooltip on hover  1536ms
         ✓ hides tooltip when mouse leaves  2486ms
         ✓ shows tooltip on keyboard focus  851ms
         ✓ hides tooltip when focus leaves  1399ms
     ✓ src/components/GlassCard.test.jsx (17 tests) 1119ms
           ✓ should pass through style prop  551ms
     ✓ src/hooks/useTheme.test.jsx (4 tests) 243ms
     ✓ src/hooks/useWasmWorker.test.js (15 tests) 313ms
     ✓ src/utils/image-utils.test.js (11 tests) 51ms
    
    ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Tests 7 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
    
     FAIL  src/components/ThemeSwitch.test.jsx > ThemeSwitch component > renders a button with the correct aria-label when theme is light
    TestingLibraryElementError: Unable to find an accessible element with the role "button" and name "Switch to Dark Mode"
    
    Here are the accessible roles:
    
      switch:
    
      Name "switch to dark mode":
      <button
        aria-checked="false"
        aria-label="switch to dark mode"
        class="glass _switch_5dee57 "
        role="switch"
        type="button"
      />
    
      --------------------------------------------------
    
    Ignored nodes: comments, script, style
    <body>
      <div>
        <div>
          <button
            aria-checked="false"
            aria-label="switch to dark mode"
            class="glass _switch_5dee57 "
            role="switch"
            type="button"
          >
            <span
              class="_thumb_5dee57"
            >
              <span
                data-testid="sun-icon"
              >
                Sun Icon
              </span>
            </span>
          </button>
        </div>
      </div>
    </body>
     ❯ Object.getElementError node_modules/@testing-library/dom/dist/config.js:37:19
     ❯ node_modules/@testing-library/dom/dist/query-helpers.js:76:38
     ❯ node_modules/@testing-library/dom/dist/query-helpers.js:52:17
     ❯ getByRole node_modules/@testing-library/dom/dist/query-helpers.js:95:19
     ❯ src/components/ThemeSwitch.test.jsx:46:27
         44|     render(<ThemeSwitch />);
         45|
         46|     const button = screen.getByRole('button', { name: 'Switch to Dark Mode' });
           |                           ^
         47|     expect(button).toBeInTheDocument();
         48|     expect(button).toHaveAttribute('type', 'button');
    
    ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/7]⎯
    
     FAIL  src/components/ThemeSwitch.test.jsx > ThemeSwitch component > shows Moon icon for light theme and applies icon class
    TestingLibraryElementError: Unable to find an element by: [data-testid="moon-icon"]
    
    Ignored nodes: comments, script, style
    <body>
      <div>
        <div>
          <button
            aria-checked="false"
            aria-label="switch to dark mode"
            class="glass _switch_5dee57 "
            role="switch"
            type="button"
          >
            <span
              class="_thumb_5dee57"
            >
              <span
                data-testid="sun-icon"
              >
                Sun Icon
              </span>
            </span>
          </button>
        </div>
      </div>
    </body>
     ❯ Object.getElementError node_modules/@testing-library/dom/dist/config.js:37:19
     ❯ node_modules/@testing-library/dom/dist/query-helpers.js:76:38
     ❯ node_modules/@testing-library/dom/dist/query-helpers.js:52:17
     ❯ getByTestId node_modules/@testing-library/dom/dist/query-helpers.js:95:19
     ❯ src/components/ThemeSwitch.test.jsx:57:29
         55|     render(<ThemeSwitch />);
         56|
         57|     const moonIcon = screen.getByTestId('moon-icon');
           |                             ^
         58|     expect(moonIcon).toBeInTheDocument();
         59|     expect(moonIcon).toHaveClass('mocked-icon-class');
    
    ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/7]⎯
    
     FAIL  src/components/ThemeSwitch.test.jsx > ThemeSwitch component > shows Sun icon for dark theme and applies icon class
    TestingLibraryElementError: Unable to find an element by: [data-testid="sun-icon"]
    
    Ignored nodes: comments, script, style
    <body>
      <div>
        <div>
          <button
            aria-checked="true"
            aria-label="switch to light mode"
            class="glass _switch_5dee57 _checked_5dee57"
            role="switch"
            type="button"
          >
            <span
              class="_thumb_5dee57"
            >
              <span
                data-testid="moon-icon"
              >
                Moon Icon
              </span>
            </span>
          </button>
        </div>
      </div>
    </body>
     ❯ Object.getElementError node_modules/@testing-library/dom/dist/config.js:37:19
     ❯ node_modules/@testing-library/dom/dist/query-helpers.js:76:38
     ❯ node_modules/@testing-library/dom/dist/query-helpers.js:52:17
     ❯ getByTestId node_modules/@testing-library/dom/dist/query-helpers.js:95:19
     ❯ src/components/ThemeSwitch.test.jsx:68:28
         66|     render(<ThemeSwitch />);
         67|
         68|     const sunIcon = screen.getByTestId('sun-icon');
           |                            ^
         69|     expect(sunIcon).toBeInTheDocument();
         70|     expect(sunIcon).toHaveClass('mocked-icon-class');
    
    ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[3/7]⎯
    
     FAIL  src/components/ThemeSwitch.test.jsx > ThemeSwitch component > calls toggleTheme when clicked
    TestingLibraryElementError: Unable to find an accessible element with the role "button" and name "Switch to Dark Mode"
    
    Here are the accessible roles:
    
      switch:
    
      Name "switch to dark mode":
      <button
        aria-checked="false"
        aria-label="switch to dark mode"
        class="glass _switch_5dee57 "
        role="switch"
        type="button"
      />
    
      --------------------------------------------------
    
    Ignored nodes: comments, script, style
    <body>
      <div>
        <div>
          <button
            aria-checked="false"
            aria-label="switch to dark mode"
            class="glass _switch_5dee57 "
            role="switch"
            type="button"
          >
            <span
              class="_thumb_5dee57"
            >
              <span
                data-testid="sun-icon"
              >
                Sun Icon
              </span>
            </span>
          </button>
        </div>
      </div>
    </body>
     ❯ Object.getElementError node_modules/@testing-library/dom/dist/config.js:37:19
     ❯ node_modules/@testing-library/dom/dist/query-helpers.js:76:38
     ❯ node_modules/@testing-library/dom/dist/query-helpers.js:52:17
     ❯ getByRole node_modules/@testing-library/dom/dist/query-helpers.js:95:19
     ❯ src/components/ThemeSwitch.test.jsx:83:27
         81|     render(<ThemeSwitch />);
         82|
         83|     const button = screen.getByRole('button', { name: 'Switch to Dark Mode' });
           |                           ^
         84|     fireEvent.click(button);
         85|
    
    ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[4/7]⎯
    
     FAIL  src/components/ThemeSwitch.test.jsx > ThemeSwitch component > is keyboard accessible (focusable)
    TestingLibraryElementError: Unable to find an accessible element with the role "button" and name "Switch to Dark Mode"
    
    Here are the accessible roles:
    
      switch:
    
      Name "switch to dark mode":
      <button
        aria-checked="false"
        aria-label="switch to dark mode"
        class="glass _switch_5dee57 "
        role="switch"
        type="button"
      />
    
      --------------------------------------------------
    
    Ignored nodes: comments, script, style
    <body>
      <div>
        <div>
          <button
            aria-checked="false"
            aria-label="switch to dark mode"
            class="glass _switch_5dee57 "
            role="switch"
            type="button"
          >
            <span
              class="_thumb_5dee57"
            >
              <span
                data-testid="sun-icon"
              >
                Sun Icon
              </span>
            </span>
          </button>
        </div>
      </div>
    </body>
     ❯ Object.getElementError node_modules/@testing-library/dom/dist/config.js:37:19
     ❯ node_modules/@testing-library/dom/dist/query-helpers.js:76:38
     ❯ node_modules/@testing-library/dom/dist/query-helpers.js:52:17
     ❯ getByRole node_modules/@testing-library/dom/dist/query-helpers.js:95:19
     ❯ src/components/ThemeSwitch.test.jsx:94:27
         92|     render(<ThemeSwitch />);
         93|
         94|     const button = screen.getByRole('button', { name: 'Switch to Dark Mode' });
           |                           ^
         95|     button.focus();
         96|     expect(button).toHaveFocus();
    
    ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[5/7]⎯
    
     FAIL  src/components/ThemeSwitch.test.jsx > ThemeSwitch component > works with unexpected/falsy theme values (defaults to Moon icon for non-"dark")
    TestingLibraryElementError: Unable to find an element by: [data-testid="moon-icon"]
    
    Ignored nodes: comments, script, style
    <body>
      <div>
        <div>
          <button
            aria-checked="false"
            aria-label="switch to dark mode"
            class="glass _switch_5dee57 "
            role="switch"
            type="button"
          >
            <span
              class="_thumb_5dee57"
            >
              <span
                data-testid="sun-icon"
              >
                Sun Icon
              </span>
            </span>
          </button>
        </div>
      </div>
    </body>
     ❯ Object.getElementError node_modules/@testing-library/dom/dist/config.js:37:19
     ❯ node_modules/@testing-library/dom/dist/query-helpers.js:76:38
     ❯ node_modules/@testing-library/dom/dist/query-helpers.js:52:17
     ❯ getByTestId node_modules/@testing-library/dom/dist/query-helpers.js:95:19
     ❯ src/components/ThemeSwitch.test.jsx:104:19
        102|     render(<ThemeSwitch />);
        103|
        104|     expect(screen.getByTestId('moon-icon')).toBeInTheDocument();
           |                   ^
        105|   });
        106|
    
    ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[6/7]⎯
    
     FAIL  src/components/ThemeSwitch.test.jsx > ThemeSwitch component > uses the toggleTheme coming from the hook
    TestingLibraryElementError: Unable to find an accessible element with the role "button" and name "Switch to Dark Mode"
    
    Here are the accessible roles:
    
      switch:
    
      Name "switch to dark mode":
      <button
        aria-checked="false"
        aria-label="switch to dark mode"
        class="glass _switch_5dee57 "
        role="switch"
        type="button"
      />
    
      --------------------------------------------------
    
    Ignored nodes: comments, script, style
    <body>
      <div>
        <div>
          <button
            aria-checked="false"
            aria-label="switch to dark mode"
            class="glass _switch_5dee57 "
            role="switch"
            type="button"
          >
            <span
              class="_thumb_5dee57"
            >
              <span
                data-testid="sun-icon"
              >
                Sun Icon
              </span>
            </span>
          </button>
        </div>
      </div>
    </body>
     ❯ Object.getElementError node_modules/@testing-library/dom/dist/config.js:37:19
     ❯ node_modules/@testing-library/dom/dist/query-helpers.js:76:38
     ❯ node_modules/@testing-library/dom/dist/query-helpers.js:52:17
     ❯ getByRole node_modules/@testing-library/dom/dist/query-helpers.js:95:19
     ❯ src/components/ThemeSwitch.test.jsx:113:24
        111|     render(<ThemeSwitch />);
        112|
        113|     const btn = screen.getByRole('button', { name: 'Switch to Dark Mode' });
           |                        ^
        114|     fireEvent.click(btn);
        115|     expect(customToggle).toHaveBeenCalledTimes(1);
    
    ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[7/7]⎯
    
    
     Test Files  1 failed | 6 passed (7)
          Tests  7 failed | 57 passed (64)
       Start at  12:46:11
       Duration  16.81s (transform 1.42s, setup 2.32s, import 4.89s, tests 18.38s, environment 17.41s)
    
    [14:46] ~/projects/Img2Num $
    ```

codevory commented:
@CodeRabbit summarize my changes

Ryan-Millard commented on src/components/GlassSwitch.jsx:
Currently `size` is not used at all. What was it intended for? Did you mean to remove it?

codevory replied:
Initially i thought i shall add this size props but now it seems it's not going to be used so yes i will be removing it .

Ryan-Millard commented on docs/docs/components/glass-switch.mdx:
Is this file still needed? I can't access it.😅

codevory replied:
i have deleted that file ( docs/docs/components/glass-switch.mdx )
Although i have improved that documentation file & moved to docs/docs/reference/react/components/GlassSwitch/index.md

codevory commented on src/components/GlassSwitch.jsx:
```diff
@@ -0,0 +1,34 @@
+      >
+        <span className={`glass ${styles.thumb} ${thumbContent ? thumbContent : fallbackContent}`}>
+          {thumbContent}
+        </span>
```
Thanks for pointing out the issue

codevory commented:
@Ryan-Millard Thank you so much for the review and kind words , I really appreciate it!

This was a great learning experience for me. The feedback around single responsibility, component architecture, and documentation quality helped me understand how real-world open-source projects are structured and maintained.

I’m especially glad the documentation met expectations . I tried to make it clear and useful for future contributors as well.
Thanks again for your guidance and for maintaining such a welcoming project. Looking forward to contributing more in the future! 🙌

Ryan-Millard commented:
> @Ryan-Millard Thank you so much for the review and kind words , I really appreciate it!
> 
> This was a great learning experience for me. The feedback around single responsibility, component architecture, and documentation quality helped me understand how real-world open-source projects are structured and maintained.
> 
> I’m especially glad the documentation met expectations . I tried to make it clear and useful for future contributors as well. Thanks again for your guidance and for maintaining such a welcoming project. Looking forward to contributing more in the future! 🙌

Hey @codevory, I’m really glad to hear that!

In a nutshell, good code aims to build **loosely-coupled, reusable components** that are simple enough to explain in a single sentence. When I was learning to code, I struggled with this too—until I read [**Programming: Principles and Practice Using C++**](https://dl.icdst.org/pdfs/files3/fef0590f02fa06bb42cba558fbc9e51c.pdf) by Bjarne Stroustrup (the creator of C++). While it isn’t directly about modern React, it covers **core concepts** like algorithms & data structures, design thinking, and decomposition & abstraction—all of which are crucial for building maintainable projects.

Regarding the documentation, you absolutely nailed it! It was clear and logically structured, which will make life easier for future contributors.

Thanks again for your contribution—I’m looking forward to seeing more from you soon! 🦔

Ryan-Millard commented:
Thank you again for the great work, @codevory. I hope you have a great day!
